src/routes/+layout.ts

```diff
@@ -20,41 +20,50 @@ export async function load() {
 	const backend = BackendService.setupInstance(queryClient);
 
 	if (browser) {
-		backend.user.fetchUserProfile().subscribe((response) => {
-			if (response.status === 'success' && response.data.state === 'SUCCESS') {
-				userProfile.set(response.data.data);
-			}
-		});
+		const fetchUserProfile = () => {
+			backend.user.fetchUserProfile().subscribe((response) => {
+				if (response.status === 'success' && response.data.state === 'SUCCESS') {
+					userProfile.update((currentProfile) => ({
+						...currentProfile,
+						...response.data.data,
+						lastUpdated: new Date()
+					}));
+				}
+			});
+		};
+
+		// Fetch user profile immediately
+		fetchUserProfile();
+
+		// Set up interval to fetch user profile
+		setInterval(fetchUserProfile, 5 * 60 * 1000);
```

Comment:
Let Tanstack query handle this. Once the stale time is reached for a query the lib will consider its data outdated and refetch if needed.